_rules/SVG-image-has-name.md

```diff
@@ -0,0 +1,141 @@
+- [ARIA10: Using aria-labelledby to provide a text alternative for non-text content](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA10)
+
+## Test Cases
+
+### Passed
+
+#### Pass example 1
+
+The `<svg>` element has a role of `img` which takes its accessible name from its `<title>` element.
+
+```html
+<svg xmlns="http://www.w3.org/2000/svg" role="img" width="100" height="100">
+  <title>A yellow circle</title>
+	<circle cx="50" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow"></circle>
+</svg>
+```
+
+#### Pass example 2
+
+The `<svg>` element has a role of `img` and an `aria-label` containing the accessible name.
+
+```html
+<svg xmlns="https://www.w3.org/2000/svg" role="img" width="100" height="100" aria-label="A yellow circle">
+	<circle cx="50" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow"></circle>
+</svg>
+```
+
+#### Pass example 3
+
+The `<svg>` element has the role of `graphics-document` which takes its accessible name from its child content, the `<text>` element.
+
+```html
+
+<svg xmlns="http://www.w3.org/2000/svg"
+     xmlns:xlink="http://www.w3.org/1999/xlink" role="graphics-document">
+    <title>3 circles</title>
+    <desc>3 yellow circles with a green stroke horizontally adjacent to each other
+    </desc>
+    <g role="group">
+	<circle cx="50" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow"></circle>
+	<circle cx="150" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow"></circle>
+	<circle cx="250" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow"></circle>
+    </g>
+</svg>
+```
+
+### Failed
+
+#### Failed example 1
+
+The `<svg>` element has a role of `img` but has no accessible name.
+
+```html
+<svg xmlns="http://www.w3.org/2000/svg" role="img" width="100" height="100">
+	<circle cx="50" cy="50" r="40" stroke="green" stroke-width="4" fill="yellow"></circle>
+</svg>
+```
+
+#### Failed example 2
+
+The `img` role includes the `<svg>` element in the accessibility tree but the element has an empty accessible name `title` attribute.
+
+```html
+<svg xmlns="http://www.w3.org/2000/svg" width="100" height="100" role="img">
+	<title> </title>
+  	<circle role="none" cx="50" cy="50" r="40" fill="yellow"></circle>
+</svg>
+```
+
+### Inapplicable
+
+#### Inapplicable example 1
+
+The `svg` element does not have a role of `img` or `graphics-document`. 
+
+```html
+<svg width="100" height="100">
+<circle cx="50" cy="50" r="40" fill="yellow"></circle>
+</svg>
+```
+
+#### Inapplicable example 2
+
+The `svg` element has an accessible name but is not included in the accessibility tree.
+
+```html
+<svg width="100" height="100" role="img" aria-label="A yellow circle" aria-hidden="true">
+<circle cx="50" cy="50" r="40" fill="yellow"></circle>
+</svg>
+```
+
+#### Inapplicable example 3
+ 
+The `svg` element is not included in the accessibility tree because the `<title>` element is empty.
+ 
+````html
+<svg xmlns="http://www.w3.org/2000/svg" width="100" height="100">
```

Comment:
I would give this example an role of img to make it absolutely clear that it is the empty title that is making it inapplicable, and not just the fact that it is not an img.

Reply:
@ShadowBB `role="img"` would make this text case applicable.

Reply:
@Brynanders I don't think so. The role="img" would not place it in the accessibility tree right? Which is what makes this inapplicable.